integration-tests/jest/jest-utils.js

```diff
@@ -16,12 +16,18 @@ const {
 
 const targetEnvParams = _.merge({ ...envs[environment] }, { billingProject, snapshotColumnName, snapshotId, snapshotTableName, testUrl, workflowName })
 
-const registerTest = ({ fn, name, timeout = defaultTimeout }) => {
-  return test(
+const registerTest = ({ fn, name, timeout = defaultTimeout, targetEnvironments = _.keys(envs) }) => {
+  return _.includes(environment, targetEnvironments) ? test(
     name,
     () => withPageLogging(withScreenshot(name)(fn))({ context, page, ...targetEnvParams }),
     timeout
-  )
+  ) : test(
+    name,
+    () => {
+      console.log(`Skipping ${name} as it is not configured to run on the ${environment} environment`)
+      new Promise(resolve => resolve())
```

Comment:
Why do you need this promise?

Reply:
Jest is expecting a promise to be returned - I will add a comment

Reply:
Actually, this is doing nothing as I am not returning it.
The error message I originally saw was: 
```    
test functions can only return Promise or undefined.
Returned value: 1
```
I had added the promise because I missed the `undefined` part of the message.
Removed the line - thanks!